### PR TITLE
make rabbit optional

### PR DESCRIPTION
**Pull request by melindaloubser1**
Added `rabbitmq.enabled` flag to influence whether rabbit related env vars and endpoints are included. This allows disabling rabbit entirely when using an alternative event broker.

### CODE REVIEW

File: charts/rasa-x/templates/event-service-deployment.yaml [hunk 1]
```diff
@@ -33,6 +33,11 @@ spec:
       # The event service uses a database, to make sure that the database is ready
       # to use we use the database migration service to check migration status.
       {{- include "initContainer.dbMigration" . |  nindent 6 }}
+      {{- if .Values.eventService.initContainers }}
+      initContainers:
```
Comment (tczekajlo) [inline charts/rasa-x/templates/event-service-deployment.yaml:37]: The problem here is that we already use a `initContainers` key in this place https://github.com/RasaHQ/rasa-x-helm/blob/master/charts/rasa-x/templates/_helpers.tpl#L267

It means if we use the db-migration service and init containers for the event service we'll have duplicate keys 🤔

**Review by tczekajlo — CHANGES_REQUESTED** [review]

File: charts/rasa-x/templates/event-service-deployment.yaml (hunk 1 above)
Comment (melindaloubser1) [reply charts/rasa-x/templates/event-service-deployment.yaml:37]: Oh dang did I add this? I didn't intend to

Comment (melindaloubser1) [reply charts/rasa-x/templates/event-service-deployment.yaml:37]: Ah, I did but it was a mistake after local experimentation, removing it

**Comment by melindaloubser1** [comment]
@tczekajlo I realize I made a false assumption here - if `rabbitmq.install` is false that doesn't mean they're not using rabbit, it could mean they're using external rabbit. Should there be a different flag, `rabbitmq.use` or `rabbitmq.use_rabbit` to turn off rabbit entirely?

**Comment by tczekajlo** [comment]
> @tczekajlo I realize I made a false assumption here - if `rabbitmq.install` is false that doesn't mean they're not using rabbit, it could mean they're using external rabbit. Should there be a different flag, `rabbitmq.use` or `rabbitmq.use_rabbit` to turn off rabbit entirely?

I think normally we use `enabled` to say if something is used or not, so maybe `rabbit.enabled`?

File: charts/rasa-x/templates/rasa-x-deployment.yaml [hunk 2]
```diff
@@ -34,6 +34,11 @@ spec:
       tolerations:
 {{ toYaml .Values.rasax.tolerations | indent 6 }}
       {{- end }}
+
+      {{- if .Values.rasax.initContainers }}
+      initContainers:
```
Comment (tczekajlo) [inline charts/rasa-x/templates/rasa-x-deployment.yaml:39]: The same problem is here, the `initContainers` is duplicated

**Review by tczekajlo — CHANGES_REQUESTED** [review]

File: charts/rasa-x/templates/rasa-x-deployment.yaml (hunk 2 above)
Comment (melindaloubser1) [reply charts/rasa-x/templates/rasa-x-deployment.yaml:39]: agh I swear I got rid of it my bad

File: charts/rasa-x/templates/rasa-x-deployment.yaml [hunk 3]
```diff
@@ -34,6 +34,11 @@ spec:
...
+{{ toYaml .Values.rasax.initContainers | indent 6 }}
+      {{- end }}
```
Comment (melindaloubser1) [inline charts/rasa-x/templates/rasa-x-deployment.yaml:41]:
```suggestion
```

**Comment by melindaloubser1** [comment]
Sorry about that, the initContainer changes were not supposed to be committed.

File: charts/rasa-x/Chart.yaml [hunk 4]
```diff
@@ -36,4 +36,4 @@ dependencies:
 - name: rabbitmq
   version: ~6.19.2
   repository: https://charts.bitnami.com/bitnami
-  condition: rabbitmq.install
+  condition: rabbitmq.install && rabbitmq.enabled
```
Comment (tczekajlo) [inline charts/rasa-x/Chart.yaml:39]:
```suggestion
  condition: rabbitmq.install
```

>Condition - The condition field holds one or more YAML paths (delimited by commas). If this path exists in the top parent's values and resolves to a boolean value, the chart will be enabled or disabled based on that boolean value. Only the first valid path found in the list is evaluated and if no paths exist then the condition has no effect.

**Review by tczekajlo — CHANGES_REQUESTED** [review]
I've left one comment. BTW, you have to bump up the chart version :)

**Review by tczekajlo — APPROVED** [review]

**Pull request merged**